### PR TITLE
Arm-ttk slash command and customization

### PR DESCRIPTION
**Pull request by v-amolpatil**
Required items, please complete
   
   Change(s):
   - Made change to existing arm-ttk-validations file and moved common code to separate file. Also added new file to do slash command using arm-ttk. Slash command will run when anyone add comment like below :'/arm-ttk', '/ARM-TTK', '/Arm-Ttk', '/armttk' or '/ARMTTK'. Also did fix deprecated command from github.

   Reason for Change(s):
   - We need to support slash command using arm-ttk.

   Version Updated:
   - NA

   Testing Completed:
   - Yes on private repo

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes

**Screenshot from private repo:**
image

### CODE REVIEW

File: .github/workflows/slash-command-armttk.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+name: Slash Command ARM-TTK Tests
+
+on:
+  issue_comment:
+    types: [created, edited]
+
+jobs:
+  run-arm-ttk:
+    runs-on: ubuntu-latest
+    if: ${{ github.actor != 'dependabot[bot]' && !github.event.pull_request.head.repo.fork && github.event.issue.pull_request && (contains(github.event.comment.body, '/armttk') || contains(github.event.comment.body, '/Armttk') || contains(github.event.comment.body, '/ARM-TTK') || contains(github.event.comment.body, '/ARMTTK') || contains(github.event.comment.body, '/arm-ttk') || contains(github.event.comment.body, '/ArmTtk')) }}
```
Comment (Tichandr) [inline .github/workflows/slash-command-armttk.yaml:11]: can't we make this array of string items and check with contains

File: .script/package-automation/arm-ttk-tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,57 @@
+
+try {
+    $diff = git diff --diff-filter=d --name-only HEAD^ HEAD
+    Write-Host "List of files in PR: $diff"
+
+    $hasmainTemplateChanged = $false
+    $hasCreateUiDefinitionTemplateChanged = $false
+
+    $isChangeInSolutionsFolder = [bool]($diff | Where-Object {$_ -like 'Solutions/*'})
```
Comment (Tichandr) [inline .script/package-automation/arm-ttk-tests.ps1:9]: What is this check for solutions folder?

Comment (v-amolpatil) [reply .script/package-automation/arm-ttk-tests.ps1:9]: It checks if there are changes in Solutions folder only. If yes then it goes and check if there are changes in maintemplate and/or createUiDefinition file.

File: .github/workflows/slash-command-armttk.yaml (hunk 1 above)
Comment (v-amolpatil) [reply .github/workflows/slash-command-armttk.yaml]: updated code

File: .script/package-automation/arm-ttk-tests.ps1 (hunk 2 above)
Comment (Tichandr) [reply .script/package-automation/arm-ttk-tests.ps1:9]: I got that. why don't we directly check inside solutions/package etc..

Comment (v-amolpatil) [reply .script/package-automation/arm-ttk-tests.ps1:9]: Inside of Solutions we have different solution folders like Alibab Cloud, Cisco Umbrella etc and inside of those folders we have Package folder. We cannot directly get Package folder for a solution. Also we need Solution Name as well.

**Review by Tichandr — APPROVED** [review]
Looks good

**Pull request merged**